### ISSUE TITLE
Support SYSDATE and SYSTIMESTAMP

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9716]
The parser is currently unaware of Oracle's `SYSDATE` and `SYSTIMESTAMP` functions. While these could naively be parsed as `DSL#currentDate()` and `DSL#currentTimestamp()` (i.e. just like `CURRENT_DATE` and `CURRENT_TIMESTAMP`) this would not always be correct. It may thus make more sense to add corresponding `DSL#systemDate()` and `DSL#systemTimestamp()` methods to the DSL API.

In some dialects like MySQL it may be possible to emulate this with something like `CONVERT_TZ(NOW(), @@session.time_zone, 'SYSTEM')`.

**Comment by lukaseder** [comment]
I agree we should fix this. I've run across this a few times myself. It's not trivial though, to fix this in a backwards compatible way. While parser support is trivial, the SQL generation might need to be adapted. I've already mentioned the risks here as well: https://github.com/jOOQ/jOOQ/issues/9715#issuecomment-573015383

**Comment by zabquart** [comment]
Any update on this? This fix would resolve having to write any query relying on sysdate as a string.

**Comment by lukaseder** [comment]
Thank you for your comment, @zabquart. I take this is related to the discussion here:
https://www.reddit.com/r/java/comments/18j46f4/comment/kdhywju/?utm_source=reddit&utm_medium=web2x&context=3

The plain SQL templating workaround that I've mentioned there can be seen here:
https://www.jooq.org/doc/latest/manual/sql-building/plain-sql-templating/
